internal/compiler/typeregister.rs

```diff
@@ -71,8 +71,77 @@ macro_rules! declare_enums {
 
 i_slint_common::for_each_enums!(declare_enums);
 
+macro_rules! map_type {
+    ($pub_type:ident, bool) => {
+        Type::Bool
+    };
+    ($pub_type:ident, i32) => {
+        Type::Int32
+    };
+    ($pub_type:ident, f32) => {
+        Type::Float32
+    };
+    ($pub_type:ident, SharedString) => {
+        Type::String
+    };
+    ($pub_type:ident, Coord) => {
+        Type::LogicalLength
+    };
+    ($pub_type:ident, KeyboardModifiers) => {
+        $pub_type.clone()
```

Comment:
btw, writting `KeyboardModifiers.clone()` directly is not working, it'll cause compile error:
> a field by this name exists in `Self`

That's why I pass in the original `$pub_type:ident` as first parameter.